var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -911,6 +1000,70 @@ def nvptx_install(self):
             make()
             make("install")
 
+    # run configure/make/make(install) for the amdgcn-amdhsa target
+    # before running the host compiler phases
+    @run_before("configure")
+    def amdgcn_install(self):
+        spec = self.spec
+        prefix = self.prefix
+
+        if not spec.satisfies("+amdgcn"):
+            return
+
+        # config.guess returns the host triple, e.g. "x86_64-pc-linux-gnu"
+        guess = Executable("./config.guess")
+        targetguess = guess(output=str).rstrip("\n")
+
+        options = getattr(self, "configure_flag_args", [])
+        options += ["--prefix={0}".format(prefix)]
+
+        # Copy LLVM utils
+        llvm_bin_path = join_path(spec["llvm"].prefix, "bin")
+        llvm_util_path = join_path(prefix, "libexec", "gcc", "amdgcn-amdhsa")
```

Comment:
I think you need to revert the llvm_util_path. Otherwise, when I try to compile it says it can't find the assembler.